bin/runtime/src/lib.rs

```diff
@@ -1277,6 +1278,17 @@ mod tests {
 
     use super::*;
 
+    #[test]
+    fn test_proxy_is_superset() {
```

Comment:
you are missing couple of asserts.

Reply:
Changed so that in `is_superset` I check all cases manually, and here I use the linear order trick.